techlibs/quicklogic/qlf_k6n10_brams.txt

```diff
@@ -0,0 +1,139 @@
+bram $__QLF_RAM16K_M0
+  init 1
+  abits 8
+  dbits 32
+  groups 2
+  ports  1  1
+  wrmode 0 1
+  enable 1 1
+  transp 0  0
+  clocks 2  3
+  clkpol 1  1
+endbram
+
+bram $__QLF_RAM16K_M1
+  init 1
+  abits  9
+  dbits  32
+  groups 2
+  ports  1 1
+  wrmode 0 1
+  enable 1 1
+  transp 0 0
+  clocks 2 3
+  clkpol 1 1
+endbram
+
```

Comment:
Should the dbits be 16 for M1 mode?

Reply:
Here's a quick explanation on what I did with the modes : 
The BRAM cells we have integrate a bit write mask ( as an input signal ). I'm using this signal to target which data bit should be written or not in the bram, dupplicating the data_in signal according to the mode (e.g: for M1 mode, we'll have din[31:16] = din[15:0] ), but we still need 32 data bits signal to read every data in the bram.
So dbits should be 32 regardless of the mode, only the address width changes, and the added address bit is used to control the bit write mask.